src/main/java/ch/wisv/areafiftylan/teams/controller/TeamRestController.java

```diff
@@ -232,6 +232,21 @@ public Team update(@PathVariable Long teamId, @Validated @RequestBody TeamDTO in
     }
 
     /**
+     * Delete the Team with the given Id. Can only be done by an admin,
+     * or the captain of the team when it has no other members.
+     *
+     * @param teamId Id of the Team to be deleted.
+     * @return A status message of the operation
+     */
+    @PreAuthorize("@currentUserServiceImpl.canDeleteTeam(principal, #teamId)")
+    @JsonView(View.Public.class)
+    @DeleteMapping("/{teamId}")
+    public ResponseEntity<?> deleteTeam(@PathVariable Long teamId) {
```

Comment:
Move the CurrentUserService Logic here. You can return a custom error response when the team is not in the right state.